client/src/components/ListItem/index.js

```diff
@@ -42,22 +44,54 @@ const ListItem = (props) => {
     });
 
   const getPosition = async () => {
-    if (!"geolocation" in navigator) {
-      console.log("Geolocation not available");
-      return [0, 0];
+    try {
+      if (!"geolocation" in navigator) {
+        console.log("Geolocation not available");
+        return [0, 0];
+      }
+      let position;
+      await getCoords().then((pos) => {
+        console.log(pos);
+        position = [pos.coords.longitude, pos.coords.latitude];
+      });
+      return position;
+    } catch (e) {
+      return [-120.54354468, 35.09349968];
     }
-    let position;
-    await getCoords().then((pos) => {
-      console.log(pos);
-      position = [pos.coords.longitude, pos.coords.latitude];
-    });
-    return position;
+  };
+
+  const getImage = (images) => {
+    if (images.length < 1) return; //invalid images array
+
+    //Otherwise, generate an objectURL using the image
+    let reader = new FileReader();
+    reader.onload = (e) => {
+      setImage(e.target.result);
+    };
+    reader.readAsDataURL(images[0]);
+    //setImage(images[0]);
   };
 
   const handleSubmit = async () => {
-    const listing = { ...fields, lngLat: await getPosition() };
     setLoading(true);
     setError(false);
```

Comment:
Is there a time where the image upload didn't work and we need to set setError to true?

Reply:
The way I have it set rn, the image is stored as a base64-encoded string directly in our MongoDB.  So the image is actually uploaded along with the actual post.  If image upload fails then the whole POST request failed, and error will be set to false.